src/borg/cache.py

```diff
@@ -612,15 +613,12 @@ def create_master_idx(chunk_idx):
             master_index_capacity = int(len(self.repository) / ChunkIndex.MAX_LOAD_FACTOR)
             if archive_ids:
                 chunk_idx = None
-                if self.progress:
```

Comment:
ProgressHandlerThingy handles that itself. Unless there is some advantage (e.g. like in "borg extract") in not providing progress information, there is no need to litter the code with "if progress: pi..." or "if pi: pi ..." anymore :)